File: 3676-smallest-number-with-all-set-bits/3676-smallest-number-with-all-set-bits.cpp
class Solution {
public:
    bool isValid(string i){
        for(auto it:i){
            if(it=='0') return false;
        }
        return true;
    }
    string binary(int n){
        string bi="";

        while(n){
            int dig=n%2;
            string digit=to_string(dig);
            n/=2;
            bi+=digit;
        }
        reverse(bi.begin(),bi.end());

        return bi;
    }
    int smallestNumber(int n) {

        for(int i=n;i<=1e6;i++){
            string bin=binary(i);
            if(isValid(bin)) return i;
        }
        return -1;
    }
};

File: 3676-smallest-number-with-all-set-bits/3676-smallest-number-with-all-set-bits.java
class Solution {
    public boolean isValid(String s){
        for(char ch:s.toCharArray()){
            if(ch=='0') return false;
        }
        return true;
    }
    public String binary(int n){
        String bi="";

        while(n>0){
            int digit=n%2;
            String dig = String.valueOf(digit);
            bi+=dig;
            n/=2;
        }
        bi = new StringBuilder(bi).reverse().toString();

        return bi;
    }
    public int smallestNumber(int n) {
        for(int i=n;i<100000;i++){
            String bi=binary(i);
            if(isValid(bi)) return i;
        }
        return -1;
    }
}